src/lib.rs

```diff
@@ -566,6 +609,95 @@ where
     }
 }
 
+// === impl GuardMut ===
+
+impl<'a, T, C: cfg::Config> GuardMut<'a, T, C> {
+    /// Returns the key used to access the guard.
+    pub fn key(&self) -> usize {
+        self.key
+    }
+}
+
+impl<'a, T, C: cfg::Config> std::ops::Deref for GuardMut<'a, T, C> {
+    type Target = T;
+
+    fn deref(&self) -> &Self::Target {
+         self.inner.item()
+    }
+}
+
+impl<'a, T, C: cfg::Config> std::ops::DerefMut for GuardMut<'a, T, C> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        self.inner.item_mut()
+    }
+}
+
+impl<'a, T, C: cfg::Config> Drop for GuardMut<'a, T, C> {
+    fn drop(&mut self) {
+        use crate::sync::atomic;
+        if self.inner.release() {
+            atomic::fence(atomic::Ordering::Acquire);
+            if Tid::<C>::current().as_usize() == self.shard.tid {
+                self.shard.remove_local(self.key);
+            } else {
+                self.shard.remove_remote(self.key);
+            }
+        }
+    }
+}
+
+impl<'a, T, C> fmt::Debug for GuardMut<'a, T, C>
+where
+    T: fmt::Debug,
+    C: cfg::Config,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt::Debug::fmt(self.inner.item(), f)
+    }
+}
+
+impl<'a, T, C> PartialEq<T> for GuardMut<'a, T, C>
+where
+    T: PartialEq<T>,
+    C: cfg::Config,
+{
+    fn eq(&self, other: &T) -> bool {
+        self.inner.item().eq(other)
+    }
+}
+
+// === impl GetMutError === 
+
+impl fmt::Display for GetMutError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self.kind {
+            GetMutErrorKind::InUse => f.pad("entry is already borrowed"),
+            GetMutErrorKind::Nonexistent => f.pad("no entry exists for the provided index"),
+        }
+    }
+}
+
+impl std::error::Error for GetMutError {}
+
+impl GetMutError {
+    pub(crate) const BORROWED: Self = Self { kind: GetMutErrorKind::InUse, };
+    pub(crate) const NONEXISTENT: Self = Self { kind: GetMutErrorKind::Nonexistent, };
+
+    pub fn is_borrowed(&self) -> bool {
```

Comment:
are there going to be more than two states? An if expression would be easier to read here imo.

```suggestion
    pub fn is_borrowed(&self) -> bool {
        self.kind == GetMutErrorKind::InUse
    }
```

Reply:
Good call, I forgot that the kind enum was `Eq`.